specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+                        string text = "Launching External Protocol";
+                        if (args.InitiatingOrigin != "")
+                        {
+                            text += "from ";
+                            text += args.InitiatingOrigin;
+                        }
+                        text += " to ";
+                        text += args.Uri;
+                        text += "\n";
+                        text += "Do you want to grant permission?";
+                        string caption = "Launching External Protocol request";
+                        MessageBoxButton btnMessageBox = MessageBoxButton.YesNoCancel;
+                        MessageBoxImage icnMessageBox = MessageBoxImage.None;
+                        MessageBoxResult resultbox = MessageBox.Show(text, caption, btnMessageBox, icnMessageBox);
+                        switch (resultbox)
+                        {
+                            case MessageBoxResult.Yes:
+                                ProcessStartInfo info = new ProcessStartInfo
+                                {
+                                    FileName = args.Uri,
+                                    UseShellExecute = true
+                                };
+                                Process.Start(info);
+                                break;
+
+                            case MessageBoxResult.No:
+                                break;
+
+                            case MessageBoxResult.Cancel:
+                                break;
+                        }
+
+                    }
+                    
+                }
+            }, null);
+        }
+    };
+}
+
+```
+
+# API Notes
+
+See [API Details](#api-details) section below for API reference.
+
+# API Details
+
+## Win32 C++
+    
+```IDL
+// This is the ICoreWebView2_16 interface.
+[uuid(cc39bea3-d6d8-471b-919f-da253e2fbf03), object, pointer_default(unique)] 
+interface ICoreWebView2_16 : ICoreWebView2_15 {
+  /// Add an event handler for the `LaunchingExternalProtocol` event.
+  /// The `LaunchingExternalProtocol` event is raised when a launch request is made to
+  /// a protocol that is registered with the OS. 
+  /// The `LaunchingExternalProtocol` event may suppress the default dialog
+  /// or replace the default dialog with a custom dialog.
+  ///
+  /// If a deferral is not taken on the event args, the external protocol launch is 
+  /// blocked until the event handler returns.  If a deferral is taken, the
+  /// external protocol launch is blocked until the deferral is completed.
+  /// The host also has the option to cancel the protocol launch.
+  ///
+  /// The `NavigationStarting` and `NavigationCompleted` events will be raised,
+  /// regardless of whether the `Cancel` property is set to `TRUE` or
+  /// `FALSE`. The `SourceChanged`, `ContentLoading`, and `HistoryChanged` events 
+  /// will not be raised regardless of this property. 

```

Comment:
Will _not_ be raised ever?

Reply:
Please update wording to make clearer that this is only for this navigation for this external URI.